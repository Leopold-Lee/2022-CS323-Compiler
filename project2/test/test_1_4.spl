struct my_struct
{
    int a;
    float b;
};

int main()
{
    int a = 0x321, b = 0X123;
    int c, d;
    c = a + b * a + b / a;
    d = (a + b) * (a + b) / a;
    if(c > 0) {
        a = 0;
    }
    if(c < 0)
    {
        if(d > 0){
            d = 1;
        }
        b = 0;
    }
    else
    {
        a = b;
    }
}