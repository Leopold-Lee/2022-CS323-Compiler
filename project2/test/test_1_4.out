Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        StructSpecifier (1)
          STRUCT
          ID: my_struct
          LC
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: int
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: a
              SEMI
            DefList (4)
              Def (4)
                Specifier (4)
                  TYPE: float
                DecList (4)
                  Dec (4)
                    VarDec (4)
                      ID: b
                SEMI
          RC
      SEMI
    ExtDefList (7)
      ExtDef (7)
        Specifier (7)
          TYPE: int
        FunDec (7)
          ID: main
          LP
          RP
        CompSt (8)
          LC
          DefList (9)
            Def (9)
              Specifier (9)
                TYPE: int
              DecList (9)
                Dec (9)
                  VarDec (9)
                    ID: a
                  ASSIGN
                  Exp (9)
                    INT: 0x321
                COMMA
                DecList (9)
                  Dec (9)
                    VarDec (9)
                      ID: b
                    ASSIGN
                    Exp (9)
                      INT: 0X123
              SEMI
            DefList (10)
              Def (10)
                Specifier (10)
                  TYPE: int
                DecList (10)
                  Dec (10)
                    VarDec (10)
                      ID: c
                  COMMA
                  DecList (10)
                    Dec (10)
                      VarDec (10)
                        ID: d
                SEMI
          StmtList (11)
            Stmt (11)
              Exp (11)
                Exp (11)
                  ID: c
                ASSIGN
                Exp (11)
                  Exp (11)
                    Exp (11)
                      ID: a
                    PLUS
                    Exp (11)
                      Exp (11)
                        ID: b
                      MUL
                      Exp (11)
                        ID: a
                  PLUS
                  Exp (11)
                    Exp (11)
                      ID: b
                    DIV
                    Exp (11)
                      ID: a
              SEMI
            StmtList (12)
              Stmt (12)
                Exp (12)
                  Exp (12)
                    ID: d
                  ASSIGN
                  Exp (12)
                    Exp (12)
                      Exp (12)
                        LP
                        Exp (12)
                          Exp (12)
                            ID: a
                          PLUS
                          Exp (12)
                            ID: b
                        RP
                      MUL
                      Exp (12)
                        LP
                        Exp (12)
                          Exp (12)
                            ID: a
                          PLUS
                          Exp (12)
                            ID: b
                        RP
                    DIV
                    Exp (12)
                      ID: a
                SEMI
              StmtList (13)
                Stmt (13)
                  IF
                  LP
                  Exp (13)
                    Exp (13)
                      ID: c
                    GT
                    Exp (13)
                      INT: 0
                  RP
                  Stmt (13)
                    CompSt (13)
                      LC
                      StmtList (14)
                        Stmt (14)
                          Exp (14)
                            Exp (14)
                              ID: a
                            ASSIGN
                            Exp (14)
                              INT: 0
                          SEMI
                      RC
                StmtList (16)
                  Stmt (16)
                    IF
                    LP
                    Exp (16)
                      Exp (16)
                        ID: c
                      LT
                      Exp (16)
                        INT: 0
                    RP
                    Stmt (17)
                      CompSt (17)
                        LC
                        StmtList (18)
                          Stmt (18)
                            IF
                            LP
                            Exp (18)
                              Exp (18)
                                ID: d
                              GT
                              Exp (18)
                                INT: 0
                            RP
                            Stmt (18)
                              CompSt (18)
                                LC
                                StmtList (19)
                                  Stmt (19)
                                    Exp (19)
                                      Exp (19)
                                        ID: d
                                      ASSIGN
                                      Exp (19)
                                        INT: 1
                                    SEMI
                                RC
                          StmtList (21)
                            Stmt (21)
                              Exp (21)
                                Exp (21)
                                  ID: b
                                ASSIGN
                                Exp (21)
                                  INT: 0
                              SEMI
                        RC
                    ELSE
                    Stmt (24)
                      CompSt (24)
                        LC
                        StmtList (25)
                          Stmt (25)
                            Exp (25)
                              Exp (25)
                                ID: a
                              ASSIGN
                              Exp (25)
                                ID: b
                            SEMI
                        RC
          RC
